subsys/net/l2/ethernet/ethernet.c

```diff
@@ -560,14 +571,17 @@ static int ethernet_send(struct net_if *iface, struct net_pkt *pkt)
 	}
 
 	ret = api->send(net_if_get_device(iface), pkt);
+
+	ethernet_remove_l2_header(pkt);
+
 	if (ret != 0) {
 		eth_stats_update_errors_tx(iface);
 		goto error;
 	}
 #if defined(CONFIG_NET_STATISTICS_ETHERNET)
 	ethernet_update_tx_stats(iface, pkt);
```

Comment:
The statistics will be incorrect now as the ethernet_update_tx_stats() will call net_pkt_get_len(pkt) to get the length. Better would be to remove the l2 header just before we return from the function.